htdocs/product/stock/stockatdate.php

```diff
@@ -582,135 +611,178 @@
+				}
+				print '</td>';
 
-			// Current stock
-			print '<td class="right">'.($currentstock ? $currentstock : '<span class="opacitymedium">0</span>').'</td>';
+				// Current stock
+				print '<td class="right">'.($currentstock ? $currentstock : '<span class="opacitymedium">0</span>').'</td>';
+			}
 			$totalcurrentstock += $currentstock;
-		}
 
-		// Fields from hook
-		$parameters = array('objp'=>$objp);
-		$reshook = $hookmanager->executeHooks('printFieldListValue', $parameters); // Note that $action and $object may have been modified by hook
-		print $hookmanager->resPrint;
+			// Fields from hook
+			$parameters = array('objp'=>$objp);
+			$reshook = $hookmanager->executeHooks('printFieldListValue', $parameters); // Note that $action and $object may have been modified by hook
+			print $hookmanager->resPrint;
 
-		// Action
-		print '<td class="right"></td>';
+			// Action
+			print '<td class="right"></td>';
 
-		print '</tr>'."\n";
+			print '</tr>'."\n";
+		}
 	}
 	$i++;
 }
 
 $parameters = array('sql'=>$sql);
 $reshook = $hookmanager->executeHooks('printFieldListFooter', $parameters); // Note that $action and $object may have been modified by hook
-print $hookmanager->resPrint;
+if ($ext!='csv') {
+	print $hookmanager->resPrint;
+}
 
 $colspan = 8;
 if ($mode == 'future') {
 	$colspan++;
 }
 
-
-if (empty($date) || !$dateIsValid) {
-	print '<tr><td colspan="'.$colspan.'"><span class="opacitymedium">'.$langs->trans("EnterADateCriteria").'</span></td></tr>';
+if ($ext=='csv') {
+	print implode("\t",
+		($mode == 'future')?array(
+		'"'.$langs->trans("Totalforthispage").'"',
+		'',
+		$productid > 0 ? "$totalcurrentstock" : '',
+		'',
+		"$totalvirtualstock"):
+		array(
+		'"'.$langs->trans("Totalforthispage").'"',
+		'',
+		'',
+		'"'.price(price2num($totalbuyingprice, 'MT')).'"',
+		!getDolGlobalString('PRODUIT_MULTIPRICES')?'"'.price(price2num($totalsellingprice, 'MT')).'"':'',
+		'',
+		$productid > 0 ? $totalcurrentstock : ''));
 } else {
-	print '<tr class="liste_total">';
-	print '<td>'.$langs->trans("Totalforthispage").'</td>';
-	print '<td></td>';
-	if ($mode == 'future') {
-		print '<td class="right">'.price(price2num($totalcurrentstock, 'MS')).'</td>';
-		print '<td></td>';
-		print '<td></td>';
-		print '<td class="right">'.price(price2num($totalvirtualstock, 'MS')).'</td>';
+	if (empty($date) || !$dateIsValid) {
+		print '<tr><td colspan="'.$colspan.'"><span class="opacitymedium">'.$langs->trans("EnterADateCriteria").'</span></td></tr>';
 	} else {
+		print '<tr class="liste_total">';
+		print '<td>'.$langs->trans("Totalforthispage").'</td>';
 		print '<td></td>';
-		print '<td class="right">'.price(price2num($totalbuyingprice, 'MT')).'</td>';
-		if (!getDolGlobalString('PRODUIT_MULTIPRICES')) {
-			print '<td class="right">'.price(price2num($totalsellingprice, 'MT')).'</td>';
+		if ($mode == 'future') {
+			print '<td class="right">'.$totalcurrentstock.'</td>';
```

Comment:
The $totalcurrentstock is a result of operation between float value so result is a float with a non exact decimal part. For this reason me must use price(price2num($totalcurrentstock, 'MS'))
price2num to trunc the decimal part that is not relevant and introduced by float number calculation. Then price to format the number into a number using the language of the user.

Reply:
This would be the same for the totalvirtualstock value as well?

Reply:
> This would be the same for the totalvirtualstock value as well? what is the difference beween the 'MS' and 'MT' value?

Yes same.MT is when we need to trunc decimal part for an amount, MS is used for a stock.

Reply:
That makes sense, I have now wrapped all the stock values after initially missing some. Hopefully this sorts it out